### PR TITLE
Implement cascade on delete/edit for model

### PR DESCRIPTION
**Pull request by uberSaiyan**

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -215,8 +227,18 @@ public void addCustomer(Customer customer) {
...
+
+        // cascade
+        List<Order> orders = orderBook.getList();
+        for (Order order : orders) {
+            if (order.getCustomer().equals(target)) {
+                Order editedOrder = new Order(order.getId(), editedCustomer, order.getPhone(),
+                        order.getPrice(), order.getStatus(), order.getSchedule(), order.getTags());
+                orderBook.setOrder(order, editedOrder);
+                break;
+            }
```
Comment (EugeneTeu) [inline src/main/java/seedu/address/model/ModelManager.java]: can right click -> use collect in intellij to convert to stream() method

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -257,6 +279,15 @@ public boolean hasPhone(Phone phone) {
...
     public void deletePhone(Phone target) {
         phoneBook.removePhone(target);
+
+        // cascade
+        List<Order> orders = orderBook.getList();
+        for (Order order : orders) {
+            if (order.getPhone().equals(target)) {
+                orderBook.removeOrder(order);
+                break;
+            }
```
Comment (EugeneTeu) [inline src/main/java/seedu/address/model/ModelManager.java]: same here

File: src/main/java/seedu/address/model/ModelManager.java [hunk 3]
```diff
@@ -73,7 +76,7 @@ public ModelManager(ReadOnlyDataBook<Customer> customerBook, ReadOnlyDataBook<Ph
                         ReadOnlyDataBook<Order> orderBook, ReadOnlyDataBook<Schedule> scheduleBook,
                         ReadOnlyUserPrefs userPrefs) {
         super();
-        requireAllNonNull(customerBook, userPrefs);
+        requireAllNonNull(customerBook, phoneBook, orderBook, scheduleBook, userPrefs);
 
         logger.fine("Initializing with customer book: " + customerBook + " and user prefs " + userPrefs);
```
Comment (qiujiaaa) [inline src/main/java/seedu/address/model/ModelManager.java]: should this be changed?

**Comment by yeodonghan** [comment]
Looks good. Great work.

**Pull request merged**